### ISSUE TITLE
Can't find migrations in deployed war on Wildfly 28

### ISSUE THREAD

**Issue by cuboci** [flyway/flyway#3667]
Flyway seems unable to scan the classpath for migration files in Wildfly 28. It worked fine in Wildfly 27(.0.1).

###### Which version and edition of Flyway are you using?

9.17.0

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)

Java API

(To be specific, I'm using the Maven plugin, too, but that applies the migrations to another database and is not relevant here.)

###### Which database are you using? (Type & version)

PostgreSQL 15.2

###### Which operating system are you using?

Debian GNU/Linux

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)

I'm packaging my web app as a war including all migration files at WEB-INF/classes/db/migration and deploying it through Maven (`mvn wildfly:deploy`). I didn't set any locations, so this is the default location. Then in my code I run migrations like this:

```java
Flyway db = Flyway.configure().dataSource(ds).load();
db.migrate();
```

The datasource is properly configured, because Wildfly reports it can see the previously applied versions.

###### What did you expect to see?

I expected new migration files to be picked up and applied.

###### What did you see instead?

I got these messages in the log:

```
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,125 INFO  [net.digide.toolbox.Database] (default task-28) location: classpath:db/migration
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,127 INFO  [org.flywaydb.core.internal.license.VersionPrinter] (default task-28) Flyway Community Edition 9.17.0 by Redgate
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,127 INFO  [org.flywaydb.core.internal.license.VersionPrinter] (default task-28) See release notes here: https://rd.gt/416ObMi
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,127 INFO  [org.flywaydb.core.internal.license.VersionPrinter] (default task-28)
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,128 WARN  [org.flywaydb.core.internal.scanner.classpath.ClassPathScanner] (default task-28) Unable to scan location: /content/toolbox.war/WEB-INF/classes/db/migration (unsupported protocol: vfs)
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,186 INFO  [org.flywaydb.core.internal.database.base.BaseDatabaseType] (default task-28) Database: jdbc:postgresql://localhost/toolbox (PostgreSQL 15.2)
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,206 INFO  [org.flywaydb.core.internal.command.DbValidate] (default task-28) Successfully validated 103 migrations (execution time 00:00.011s)
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,221 INFO  [org.flywaydb.core.internal.command.DbMigrate] (default task-28) Current version of schema "public": 0105
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,221 ERROR [org.flywaydb.core.internal.command.DbMigrate] (default task-28) Schema "public" has version 0105, but no migration could be resolved in the configured locations !
May 03 15:13:04 gamont standalone.sh[634855]: 15:13:04,221 INFO  [org.flywaydb.core.internal.command.DbMigrate] (default task-28) Schema "public" is up to date. No migration necessary.
```

This worked fine in Wildfly 27. Downgrading and deploying the app on it picked up the new migrations (version 0106 and 0107) and applied them.

**Comment by JasonLuo-Redgate** [comment]
Hi @cuboci , based on the provide log
Unable to scan location: /content/toolbox.war/WEB-INF/classes/db/migration (unsupported protocol: vfs)
I think this may highly related to:
 https://www.wildfly.org/news/2023/04/20/WildFly28-Released/ 
https://issues.redhat.com/browse/WFLY-17921
Could you try the workaround mentioned in it?

**Comment by cuboci** [comment]
Hi @JasonLuo-Redgate, thanks for looking into this. Adding that workaround indeed fixes the problem. However, I'm a little wary of doing it like this because, as the issue mentioned, this should not be considered a fix. Is this something that needs to be fixed in Flyway or Wildfly?

**Comment by JasonLuo-Redgate** [comment]
Hi @cuboci , this is marked as a known issue in WildFly28, and as mentioned in the release notes, this should be resolved in WildFly 28.0.1.
